drivers: peci: xec: Handle PECI command PING properly

JMasterMeister opened the pull request:
Handle PECI command PING properly, also get Write FCS Byte as a check.
Now it is possible to perform a PECI Ping without crashing the bus
and a check whether the Ping was sucessful or not through Write FCS.

Signed-off-by: Johannes Meister <johannes.meister@kontron.com>

JMasterMeister committed:
drivers: peci: xec: Handle PECI command PING properly

Handle PECI command PING properly, also get Write FCS Byte as a check.
Now it is possible to perform a PECI Ping without crashing the bus or
blocking it for subsequent PECI transactions.
It is also possible to check whether the Ping was sucessful
or not with the Write FCS Byte.

Signed-off-by: Johannes Meister <johannes.meister@kontron.com>

albertofloyd commented:
@JMasterMeister could you also update samples/drivers/peci with test code that causes crash? 

This will allow to catch similar issues in the future.

JMasterMeister commented on drivers/peci/peci_mchp_xec.c:
```diff
@@ -250,18 +252,38 @@ static int peci_xec_read(const struct device *dev, struct peci_msg *msg)
+			if (i == 0) {
+				/* Get write block FCS just for debug */
+				tx_fcs = base->RD_DATA;
+				LOG_DBG("TX FCS %x", tx_fcs);
```
@VenkatKotakonda, yes you are right. This change is tested and works as intended.
A patch including this changes follows soon.

JMasterMeister commented:
@albertofloyd to reproduce this issue the existing peci sample can be used. When the function "peci_ping() and a
subsequent peci function like peci_get_tjmax() is called, logging api shows invalid peci response. 
It seems that there are some problems with the content of the receive registers in MEC hardware,
maybe old data from peci_ping() command interferes with the subsequent incoming data.

JMasterMeister commented:
@albertofloyd have you already checked or reproduced this issue?

albertofloyd commented:
@JMasterMeister see draft of samples/drivers/peci update https://github.com/zephyrproject-rtos/zephyr/pull/44436
Does this changes prevents the bus from going into recovery? 

R FCS 4c
Temp bytes: eaed
Temperature 27 C
peci_ping
**[00:00:07.577,087] <wrn> peci_mchp_xec: peci_xec_bus_recovery full_reset:0
[00:00:07.582,153] <wrn> peci_mchp_xec: peci_xec_bus_recovery full_reset:0**
R FCS 9a
Temp bytes: e4ed
Temperature 27 C
peci_ping
**[00:00:08.582,672] <wrn> peci_mchp_xec: peci_xec_bus_recovery full_reset:0
[00:00:08.587,738] <wrn> peci_mchp_xec: peci_xec_bus_recovery full_reset:0**
R FCS 27
Temp bytes: eded
Temperature 27 C

JMasterMeister commented:
@albertofloyd yes, this should prevent from bus recovery. For my special implementation, a new command after a ping additionally returns an invalid peci response on a rdpkgconfig command. But maybe this is an additional issue from my implementation.

JMasterMeister commented:
How is it possible to get over the Compliance Checks which failed for this case? I changed my user name to fulfill the requirements and also changed the commit message but this fail still turns out.
I can not find the message anymore where this shows up:

9d0fffc913adec5f690eea0c6b58d9ae45b130da: author email (johannesmeister <johannes.meister@kontron.com>) needs to match one of the signed-off-by entries.
9d0fffc913adec5f690eea0c6b58d9ae45b130da: author email (johannesmeister <johannes.meister@kontron.com>) does not follow the syntax: First Last <email>.
Error: Process completed with exit code 1.

albertofloyd commented:
> johannesmeister
@JMasterMeister Try using first, last in your name i.e. 
Signed-off-by: Jose Alberto Meza <jose.a.meza.arellano@intel.com>

dcpleung commented on drivers/peci/peci_mchp_xec.c:
```diff
@@ -296,8 +318,9 @@ static int peci_xec_transfer(const struct device *dev, struct peci_msg *msg)
 
 	/* If a PECI transmission is successful, it may or not involve
 	 * a read operation, check if transaction expects a response
+	 * Also perform a read when PECI cmd is Pingt to get Write FCS
```
Is `Pingt` the actual command name?

JMasterMeister replied:
@dcpleung no, this is a typo. I will rework it

dcpleung commented:
Might want to try `git commit --amend --author="Johannes Meister <johannes.meister@kontron.com>"` on all the comments to get past the compliance check error.

JMasterMeister commented:
@dcpleung The only location I found the author name like it is shown in the error check is in the comment for this pull request. I changed the author string to the required format. Let's see if it works now :)

dcpleung commented:
Doesn't seem to have fixed it... if you run `git log --format=full` you can see the `Author` field in the commits.

And also, please rebase to remove the merge commit.

dcpleung commented:
I pulled the PR and the author field of commit `drivers: peci: xec: Handle PECI command PING properly` still has `johannesmeister` as name.

JMasterMeister commented:
@dcpleung, I rebased the branch on latest main branch and also did the interactive rebase to change the author field of the first commit. Searching for the "wrong author" in log was successfully before the rebase, after rebase no commit with old author name could be found. So hopefully this time it worked as it should.
By the way, thanks for your help :)

MaureenHelm commented:
This patch series is difficult to follow. It looks like all but the first commit are fixup commits to address issues that arose during code review and CI. If that's the case, please squash the fixup commits to make it look like you did the right thing on the first try.

carlescufi commented:
@jvasanth1 could you please take a look and approve if happy?
